fetcher/src/job/job.utils.ts

```diff
@@ -227,31 +230,31 @@ export function extractFeeds(
  * submission more than given threshold or absolute threshold. If yes,
  * return `true`, otherwise `false`.
  *
- * @param {number} latest submission value
- * @param {number} current submission value
- * @param {number} threshold configuration
- * @param {number} absolute threshold configuration
- * @return {boolean}
+ * @param {number} aggregatorAddress - Aggregator contract address
+ * @param {number} currentSubmission - Submission value
+ * @param {number} threshold - Threshold configuration
+ * @param {number} absoluteThreshold - AbsoluteThreshold configuration
+ * @return {boolean} Result in boolean, True or False
  */
-export function shouldReport(
-  latestSubmission: number,
-  submission: number,
-  decimals: number,
+
+export async function shouldReport(
+  aggregatorAddress: string,
+  currentSubmission: number,
   threshold: number,
   absoluteThreshold: number
-): boolean {
-  if (latestSubmission && submission) {
-    const denominator = Math.pow(10, decimals)
-    const latestSubmissionReal = latestSubmission / denominator
-    const submissionReal = submission / denominator
+): Promise<boolean> {
+  const contract = new ethers.Contract(aggregatorAddress, Aggregator__factory.abi, PROVIDER)
+  const latestSubmission = Number((await contract.latestRoundData()).answer)
```

Comment:
I think it would be better to reuse the value returned from `oracleRoundState` call. It includes the last submission of the requested oracle (aka reporter)

https://github.com/Bisonai/orakl/blob/3eaa4c11ec1f3ff8088a84477c0076adac103cae/contracts/src/v0.1/Aggregator.sol#L356

which is a better description of deviation, otherwise this deviation could be triggered much more often than we want since there are is no guarantee that two or more reporters will always report values close to each other. If they are differ always more than relative threshold, deviation worker will start triggering new rounds every two seconds leading to undesired update speed.

Additionally, we can save one call to JSON-RPC which is good because this job runs quite often. We will be able to save 16 calls per second to our EN node.